src/csharp/NativeMethods.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntimeGenAI
+{
+    internal static class NativeMethods
+    {
+        internal class NativeLib
+        {
+            internal const string DllName = "onnxruntime-genai";
+        }
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern IntPtr /* const char* */ OgaResultGetError(IntPtr /* OgaResult* */ result);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern void OgaDestroyResult(IntPtr /* OgaResult* */ result);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern IntPtr /* OgaResult* */ OgaCreateModel(string /* const char* */ configPath,
```

Comment:
You can not use string, because it will not be converted to ANSI by default, not UTF-8. In this case, the client code must covert the string to UTF-8 `byte[]`. Then you can pass it as such.

Reply:
Addressed this.